### DISCUSSION TITLE
what's the best way to conditionally loads one lsp server over another (typescript, angular, html)

### DISCUSSION THREAD

**Discussion by IzioDev** [NvChad/NvChad#2022] [Q&A] [answered]
Hi,

I have a test project, where i'm leaning `nvim` on it, it is a monorepo containing multiple framework / languages, which incluces:
* typescript (server)
* angular (client)
* basic html template

I wanted to know the idiomatic way of conditionally loads on server over another, currently, `typescript` is loaded on the angular project on `*.component.ts` files, and `html` is loaded on `*.component.html` files.

I though about see if the file match a certain pattern (`component|module|routes`), but it might actually conflicts with other project filenames.

I also would like to not interfer with any project-wide configuration, if possible.

How are you dealing with these things ?

PS: I'm new, so don't hesitate to over detail the solution

**Comment by siduck** [comment, marked as answer]
add filetypes option in your lspconfig lang setup calls like

```lua
lspconfig.html.setup {
   filetypes = { "html" }
}
```

**Reply by IzioDev** [reply]
hi, thank you for you reply @siduck 

Does this means, `html` will loads on every `.html` files ?

wouldn't that interfer with `angular-html` files ?

Thanks in advance

**Reply by siduck** [reply]
@IzioDev you can have multiple lsps defined for a filetype, btw .html wont interfere with angular-html
